src/components/textField/index.tsx

```diff
@@ -135,33 +139,38 @@ const TextField = (props: InternalTextFieldProps) => {
   return (
     <FieldContext.Provider value={context}>
       <View {...containerProps} style={[margins, positionStyle, containerStyle, centeredContainerStyle]}>
-        <Label
-          label={label}
-          labelColor={labelColor}
-          labelStyle={_labelStyle}
-          labelProps={labelProps}
-          floatingPlaceholder={floatingPlaceholder}
-          validationMessagePosition={validationMessagePosition}
-          testID={`${props.testID}.label`}
-          showMandatoryIndication={showMandatoryIndication}
-          enableErrors={enableErrors}
-        />
-        {validationMessagePosition === ValidationMessagePosition.TOP && (
-          <ValidationMessage
+        <View row spread>
+          <Label
+            label={label}
+            labelColor={labelColor}
+            labelStyle={_labelStyle}
+            labelProps={labelProps}
+            floatingPlaceholder={floatingPlaceholder}
+            validationMessagePosition={validationMessagePosition}
+            testID={`${props.testID}.label`}
+            showMandatoryIndication={showMandatoryIndication}
             enableErrors={enableErrors}
-            validate={others.validate}
-            validationMessage={others.validationMessage}
-            validationMessageStyle={_validationMessageStyle}
-            retainValidationSpace={retainValidationSpace && retainTopMessageSpace}
-            testID={`${props.testID}.validationMessage`}
           />
-        )}
+          {validationMessagePosition === ValidationMessagePosition.TOP && (
+            <ValidationMessage
+              enableErrors={enableErrors}
+              validate={others.validate}
+              validationMessage={others.validationMessage}
+              validationMessageStyle={_validationMessageStyle}
+              retainValidationSpace={retainValidationSpace && retainTopMessageSpace}
+              testID={`${props.testID}.validationMessage`}
+            />
+          )}
+          <View marginL-s2>
```

Comment:
The `marginL-s2` is to give a minimum space btw the accessory and the label, as per the requirement in the guidelines. 
If I leave it up to the user who passes the accessory it might look bad with long labels (If I'm removing it I'll remove the View wrapper altogether...)

Reply:
Yes, I understand. it just forcing this spacing sounds like a bad idea, because you are not leaving the users a way to change that. And I don't think we'll have a container style for this specific view anytime soon. 
I'm fine with moving this responsibility to the usage. 
Also, now you add a margin whether the user passed `topTrailingAccessory` or not, obviously it's easier to fix. But let's remove it all together.

Reply:
fixed